mmv1/products/compute/NetworkEndpoints.yaml

```diff
@@ -0,0 +1,140 @@
+identity:
+  - networkEndpointGroup
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/load-balancing/docs/negs/'
+  api: 'https://cloud.google.com/compute/docs/reference/rest/beta/networkEndpointGroups'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    kind: 'compute#operation'
+    path: 'name'
+    base_url: 'projects/{{project}}/zones/{{zone}}/operations/{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'targetLink'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'status'
+    complete: 'DONE'
+    allowed:
+      - 'PENDING'
+      - 'RUNNING'
+      - 'DONE'
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error/errors'
+    message: 'message'
+id_format: '{{project}}/{{zone}}/{{network_endpoint_group}}/endpoints'
+import_format:
+  - 'projects/{{project}}/zones/{{zone}}/networkEndpointGroups/{{network_endpoint_group}}'
+mutex: networkEndpoint/{{project}}/{{zone}}/{{network_endpoint_group}}
+exclude_validator: true
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_endpoints'
+    primary_resource_id:
+      'default-endpoints'
+      # Fine-grained resource need different autogenerated tests, as
+      # we need to check destroy during a test step where the parent resource
+      # still exists, rather than during CheckDestroy (when read returns
+      # nothing because the parent resource has then also been destroyed)
+    skip_test: true
+    vars:
+      neg_name: 'my-lb-neg'
+      instance_name: 'endpoint-instance'
+      network_name: 'neg-network'
+      subnetwork_name: 'neg-subnetwork'
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  constants: templates/terraform/constants/network_endpoints.go.erb
+  pre_create: templates/terraform/pre_create/network_endpoints.go.erb
+  pre_update: templates/terraform/pre_update/network_endpoints.go.erb
+  pre_delete: templates/terraform/pre_delete/compute_network_endpoints.go.erb
+  decoder: templates/terraform/decoders/network_endpoints.go.erb
+  encoder: templates/terraform/encoders/compute_network_endpoints.go.erb
+parameters:
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'zone'
+    resource: 'Zone'
+    imports: 'name'
+    description: |
+      Zone where the containing network endpoint group is located.
+    required: false
+    url_param_only: true
+    default_from_api: true
+    ignore_read: true
+  - !ruby/object:Api::Type::ResourceRef
+    name: 'networkEndpointGroup'
+    resource: 'NetworkEndpointGroup'
+    imports: 'name'
+    description: |
+      The network endpoint group this endpoint is part of.
+    required: true
+    url_param_only: true
+    ignore_read: true
+    diff_suppress_func: tpgresource.CompareResourceNames
+properties:
+  - !ruby/object:Api::Type::Array
+    name: 'endpoints'
```

Comment:
So, normally we try to make the names of the fields in the terraform resource match the field name in the API. In this case that would mean renaming this field to `networkEndpoints` (which would let you remove the `api_name` override). It does feel a little duplicative here since the resource is already called network_endpoints, but in general the closer you can be to the API the more predictable the resource is for users and the easier to maintain in the long run.

If you have strong feelings about keeping it "endpoints", I won't block on this; it's just my recommendation.

Reply:
No strong feelings about `endpoints`, I just thought it was less clunky. Changed to `networkEndpoints`.